.github/workflows/site.yaml

```diff
@@ -0,0 +1,33 @@
+name: Site
+
+on:
+  push:
+    branches:
+      - site
+
+jobs:
+  build:
+    if: github.repository_owner == 'oshi' && ! contains(toJSON(github.event.head_commit.message), '[maven-release-plugin]')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - uses: webfactory/ssh-agent@master
+        with:
+          ssh-private-key: ${{ secrets.DEPLOY_KEY }}
+      - name: Build site
+        run: ./mvnw site site:stage -DskipTests
+        env:
+          CI_DEPLOY_USERNAME: ${{ secrets.CI_DEPLOY_USERNAME }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Deploy Site to gh-pages
```

Comment:
How often are we deploying new javadocs?

Reply:
Probably not often, this is set to run only if we kick a 'site' branch.  It otherwise is released separately when you perform the release cycle.  A few other projects I work on folks wanted those automated so I figured might as well play around and see if it is feasable.  If we start here seeing the automation at work, it becomes easier to see that we could revamp the repo and let github take care of the full release.